zilliqa/src/consensus.rs

```diff
@@ -1119,7 +1131,7 @@ impl Consensus {
             proposal.header.transactions_root_hash,
             proposal.header.receipts_root_hash,
             proposal.transactions,
-            proposal.header.timestamp,
+            SystemTime::max(SystemTime::now(), parent_block.timestamp()),
```

Comment:
I tried doing this also, it seems transactions use the block timestamp when executing so we cannot change it here after having executed transactions for this block. See: https://github.com/Zilliqa/zq2/pull/1606

Reply:
Noted. Will revert.